packages/flutter/test/material/floating_action_button_location_test.dart

```diff
@@ -609,8 +609,749 @@ void main() {
+      const Rect defaultRect = Rect.fromLTRB(376.0, 506.0, 424.0, 554.0);
+      // Positioned relative to BottomSheet
+      const Rect bottomSheetRect = Rect.fromLTRB(376.0, 480.0, 424.0, 528.0);
+      // Positioned relative to SnackBar
+      const Rect snackBarRect = Rect.fromLTRB(376.0, 498.0, 424.0, 546.0);
+      await _runFloatTests(
+        tester,
+        FloatingActionButtonLocation.miniCenterFloat,
+        defaultRect,
+        bottomSheetRect,
+        snackBarRect,
+        mini: true,
+      );
+    });
+
+    testWidgets('endFloat', (WidgetTester tester) async {
+      const Rect defaultRect = Rect.fromLTRB(728.0, 494.0, 784.0, 550.0);
+      // Positioned relative to BottomSheet
+      const Rect bottomSheetRect = Rect.fromLTRB(728.0, 472.0, 784.0, 528.0);
+      // Positioned relative to SnackBar
+      const Rect snackBarRect = Rect.fromLTRB(728.0, 486.0, 784.0, 542.0);
+      await _runFloatTests(
+        tester,
+        FloatingActionButtonLocation.endFloat,
+        defaultRect,
+        bottomSheetRect,
+        snackBarRect,
+      );
+    });
+
+    testWidgets('miniEndFloat', (WidgetTester tester) async {
+      const Rect defaultRect = Rect.fromLTRB(740.0, 506.0, 788.0, 554.0);
+      // Positioned relative to BottomSheet
+      const Rect bottomSheetRect = Rect.fromLTRB(740.0, 480.0, 788.0, 528.0);
+      // Positioned relative to SnackBar
+      const Rect snackBarRect = Rect.fromLTRB(740.0, 498.0, 788.0, 546.0);
+      await _runFloatTests(
+        tester,
+        FloatingActionButtonLocation.miniEndFloat,
+        defaultRect,
+        bottomSheetRect,
+        snackBarRect,
+        mini: true,
+      );
+    });
+
+    // Test docked locations, for each (6), keyboard presented or not:
+    //  - Default
+    //  - Default with resizeToAvoidBottomInset: false
+    //  - docked with BottomNavigationBar
+    //  - docked with BottomNavigationBar and resizeToAvoidBottomInset: false
+    //  - docked with BottomNavigationBar & BottomSheet
+    //  - docked with BottomNavigationBar & BottomSheet, resizeToAvoidBottomInset: false
+    //  - with SnackBar
+    Future<void> _runDockedTests(
+      WidgetTester tester,
+      FloatingActionButtonLocation location,
+      Rect defaultRect,
+      Rect bottomNavigationBarRect,
+      Rect bottomSheetRect,
+      Rect snackBarRect, {
+      bool mini = false,
+    }) async  {
+      const double keyboardHeight = 200.0;
+      const double viewPadding = 50.0;
+      const double bottomNavHeight = 99.0;
+      final Key floatingActionButton = UniqueKey();
+      final double fabHeight = mini ? 48.0 : 56.0;
+      // Default
+      await tester.pumpWidget(_buildTest(
+        location,
+        const MediaQueryData(viewPadding: EdgeInsets.only(bottom: viewPadding)),
+        floatingActionButton,
+        mini: mini,
+      ));
+      expect(
+        tester.getRect(find.byKey(floatingActionButton)),
+        rectMoreOrLessEquals(defaultRect),
```

Comment:
Whey are we using rectMoreOrlessEquals here and elsewhere? It would be OK to just include an explanatory comment.

Reply:
Oh I borrowed a little from other FAB tests. I'm not sure we have to use `rectMoreOrLessEquals`. I'll see if I can simplify.

Reply:
Ah, just checked. This is required as it accounts for the precisionErrorTolerance. The tests will not pass otherwise.